### PR TITLE
Fix typo in `EarlyStopping` error message.

### PR DESCRIPTION
**Pull request by nijkah**
## What does this PR do?

Fix typo in `EarlyStopping` error message.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by nijkah** [comment]
In addition to this,
I wonder that we should update `self.best_score` to `current` when the condition is rechead.
```
        elif self.stopping_threshold is not None and self.monitor_op(current, self.stopping_threshold):
            should_stop = True
            reason = (
                "Stopping threshold reached:"
                f" {self.monitor} = {current} {self.order_dict[self.mode]} {self.stopping_threshold}."
                " Signaling Trainer to stop."
            )
```

File: src/pytorch_lightning/callbacks/early_stopping.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def _validate_condition_metric(self, logs: Dict[str, Tensor]) -> bool:
         monitor_val = logs.get(self.monitor)
 
         error_msg = (
-            f"Early stopping conditioned on metric `{self.monitor}` which is not available."
+            f"Early stopping conditioned on metric `{self.monitor}` is not available."
```
Comment (awaelchli) [inline src/pytorch_lightning/callbacks/early_stopping.py:144]: I don't think this change is correct. We are not trying to say that "early stopping [...] is not available". We are pointing to the metric not being available. Was there a misunderstanding?

Comment (nijkah) [reply src/pytorch_lightning/callbacks/early_stopping.py:144]: OMG. Yeah, you are right. I hurried before looking into it. 🥲 
I apologize for my silly PR.

Comment (awaelchli) [reply src/pytorch_lightning/callbacks/early_stopping.py:144]: No problem at all!

**Pull request closed without merging**